sdk/core/Azure.Core/src/ResponseOfT.cs

```diff
@@ -20,7 +20,10 @@ public abstract class Response<T> : NullableResponse<T>
     {
         /// <inheritdoc />
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public override bool HasValue => true;
+        public override bool HasValue => Value != null;
+
+        /// <inheritdoc />
+        public override T Value => Value!;
```

Comment:
Should this throw if Value is null?  Or do we have a strict guarantee that this can never be true?

Reply:
This was [discussed a bit](https://github.com/Azure/azure-sdk-for-net/issues/35475#issuecomment-1505972095), and I think we are leaning towards not throwing.

Reply:
Thinking more about this ... there was a question regarding how we missed making Value nullable in the initial type release.  Looking at the [tests in that PR](https://github.com/Azure/azure-sdk-for-net/pull/31790/files#diff-94be93ea9d9cac57dbbc43d3e1f78f579e66881b08a069274281b017ff6fb9f4R29) and the [implementation of Nullable\<T\>](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L38), my guess is that we copied the Nullable\<T\> implementation with the idea that any subtype of NullableResponse\<T\> that could have a null value would throw if someone tried to access the value when one hadn't been assigned.

I know we can't throw now without risking breaking changes, but without nullability enabled in client libraries, is there a way we could use Azure.Core's nullability constraint to enforce that any new operations returning Response\<T\> can't actually have a null value?

Reply:
Would it make sense to add a Debug.Assert in Response.FromValue(), or the `ValueResponse<T>` constructor if value is passed in as null?  Or would this constitute a breaking change?  I'm just wondering if there are still ways we can set SDK authors up to follow intended guidance without introducing breaking changes.

Reply:
Yes, I believe that would break any library tests that are currently setting null here. I'm not sure if there is a great way to steer SDK authors away from ever returning null here without breaking them. I'm open to ideas though.